### PR TITLE
Merge beta RDAP feature for production release

### PR DESCRIPTION
**Pull request by eshryane**
To get started, have a look at the WhoisRdapService.java (the entry-point service for the REST API) and WhoisRdapServiceTestIntegration.java (lots of integration tests)

### CODE REVIEW

File: whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCInfoDecorator.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 
 import net.ripe.db.whois.common.collect.IterableTransformer;
+import net.ripe.db.whois.common.domain.CIString;
```
Comment (dacruz) [inline whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCInfoDecorator.java]: Let's remove unnecessary imports. 
Below as well...

File: whois-query/src/test/java/net/ripe/db/whois/query/integration/WhoisShutdownTestIntegration.java [hunk 2]
```diff
@@ -37,6 +38,7 @@ public void tearDown() throws Exception {
     }
 
     @Test
+    @Ignore //TODO unignore when rdap branch goes into master
```
Comment (dacruz) [inline whois-query/src/test/java/net/ripe/db/whois/query/integration/WhoisShutdownTestIntegration.java]: Is it time to unignore?

File: whois-query/src/test/java/net/ripe/db/whois/query/planner/AbuseCInfoDecoratorTest.java [hunk 3]
```diff
@@ -13,6 +13,7 @@
 import org.mockito.runners.MockitoJUnitRunner;
 
 import java.util.Collections;
+import java.util.HashMap;
```
Comment (dacruz) [inline whois-query/src/test/java/net/ripe/db/whois/query/planner/AbuseCInfoDecoratorTest.java]: Unused import.

File: whois-rpsl/src/main/java/net/ripe/db/whois/common/ip/Ipv6Resource.java [hunk 4]
```diff
@@ -255,9 +255,24 @@ public int compareUpperBound(final Ipv6Resource that) {
...
-        System.arraycopy(Longs.toByteArray(beginMsb), 0, bytes, 8, 8);
-        System.arraycopy(Longs.toByteArray(beginLsb), 0, bytes, 0, 8);
+        byte[] bytes = new byte[16];
+
+        System.arraycopy(Longs.toByteArray(beginMsb), 0, bytes, 0, 8);
+        System.arraycopy(Longs.toByteArray(beginLsb), 0, bytes, 8, 8);
+        try {
+            return Inet6Address.getByAddress(bytes);
+        } catch (UnknownHostException e) {
+            // this will never happen
```
Comment (dacruz) [inline whois-rpsl/src/main/java/net/ripe/db/whois/common/ip/Ipv6Resource.java]: Should we at least log it? DEBUG?

File: whois-rpsl/src/test/java/net/ripe/db/whois/common/ip/IpIntervalTest.java [hunk 5]
```diff
@@ -1,5 +1,6 @@
 package net.ripe.db.whois.common.ip;
 
+import net.ripe.db.whois.common.ip.IpInterval;
```
Comment (dacruz) [inline whois-rpsl/src/test/java/net/ripe/db/whois/common/ip/IpIntervalTest.java]: Another unused import

File: whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/attrs/ChangedTest.java [hunk 6]
```diff
@@ -1,5 +1,7 @@
 package net.ripe.db.whois.common.rpsl.attrs;
 
+import net.ripe.db.whois.common.rpsl.attrs.AttributeParseException;
```
Comment (dacruz) [inline whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/attrs/ChangedTest.java]: Another unused import

File: whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/attrs/DsRdataTest.java [hunk 7]
```diff
@@ -1,5 +1,6 @@
 package net.ripe.db.whois.common.rpsl.attrs;
 
+import net.ripe.db.whois.common.rpsl.attrs.DsRdata;
```
Comment (dacruz) [inline whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/attrs/DsRdataTest.java]: Another unused import

File: whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/attrs/MntRoutesTest.java [hunk 8]
```diff
@@ -1,5 +1,7 @@
 package net.ripe.db.whois.common.rpsl.attrs;
 
+import net.ripe.db.whois.common.rpsl.attrs.AttributeParseException;
```
Comment (dacruz) [inline whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/attrs/MntRoutesTest.java]: Another unused import

File: whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/domain/Notice.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+    public List<Link> getLinks() {
+        if (links == null) {
+            links = Lists.newArrayList();
+        }
+        return links;
+    }
+
+    @Override
+    public int compareTo(Notice o) {
```
Comment (petchells) [inline whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/domain/Notice.java]: missing 'final'

File: whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/domain/Notice.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+public class Notice implements Serializable, Comparable<Notice> {
+    protected String title;
+    protected List<String> description;
+    protected List<Link> links;
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(String value) {
```
Comment (petchells) [inline whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/domain/Notice.java]: missing 'final'

File: whois-rpsl/src/main/java/net/ripe/db/whois/common/ip/Ipv6Resource.java (hunk 4 above)
Comment (eshryane) [inline whois-rpsl/src/main/java/net/ripe/db/whois/common/ip/Ipv6Resource.java]: No need to log it, as we always return null on error

File: whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/WhoisRdapService.java [hunk 11]
```diff
@@ -0,0 +1,506 @@
...
+import static net.ripe.db.whois.common.rpsl.ObjectType.ORGANISATION;
+import static net.ripe.db.whois.common.rpsl.ObjectType.PERSON;
+import static net.ripe.db.whois.common.rpsl.ObjectType.ROLE;
+
+@Component
+@Path("/")
+public class WhoisRdapService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WhoisRdapService.class);
+    private static final int STATUS_TOO_MANY_REQUESTS = 429;
+    private static final Set<ObjectType> ABUSE_CONTACT_TYPES = Sets.newHashSet(AUT_NUM, INETNUM, INET6NUM);
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/WhoisRdapService.java]: Fixed

File: whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/domain/Notice.java (hunk 10 above)
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/domain/Notice.java]: Fixed

File: whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/domain/Notice.java (hunk 9 above)
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/whois/rdap/domain/Notice.java]: Fixed

File: whois-query/src/test/java/net/ripe/db/whois/query/integration/WhoisShutdownTestIntegration.java (hunk 2 above)
Comment (eshryane) [inline whois-query/src/test/java/net/ripe/db/whois/query/integration/WhoisShutdownTestIntegration.java]: Fixed

File: whois-client/src/test/java/net/ripe/db/whois/query/QueryFlagTest.java [hunk 12]
```diff
@@ -1,5 +1,6 @@
 package net.ripe.db.whois.query;
 
+import net.ripe.db.whois.query.QueryFlag;
```
Comment (eshryane) [inline whois-client/src/test/java/net/ripe/db/whois/query/QueryFlagTest.java]: Not rdap specific, I'll fix on master.

Comment (eshryane) [inline whois-client/src/test/java/net/ripe/db/whois/query/QueryFlagTest.java]: Fixed in commit e7bef883a6354406edf7ba7d7102a8a9b27ec9c7

**Pull request merged**